src/rules/jsxNamedPropInterfaceRule.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Lint from "tslint";
+import { isInterfaceDeclaration } from "tsutils";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public static FAILURE_MESSAGE = `interface name must be $\{componentName\}Props`;
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    return this.applyWithFunction(sourceFile, walk);
+  }
+}
+
+function walk(ctx: Lint.WalkContext<void>) {
+  return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
+    const isTsx = ctx.sourceFile.languageVariant === 1;
+    if (isInterfaceDeclaration(node) && isTsx) {
```

Comment:
What if we want to declare an interface in `.tsx` which is not a prop-types; e.g. state, something else entirely?

Reply:
yeah, I have the same feedback. I don't think this rule is generalizable in its current state. You should try using it as a custom rule in your own repo, then writing a proposal for a new rule you'd like to add to tslint-react, and then submitting a PR like this